src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateUtils.java

```diff
@@ -407,6 +412,8 @@ public static PodTemplate combine(PodTemplate parent, PodTemplate template) {
         podTemplate.setRunAsUser(template.getRunAsUser() != null ? template.getRunAsUser() : parent.getRunAsUser());
         podTemplate.setRunAsGroup(template.getRunAsGroup() != null ? template.getRunAsGroup() : parent.getRunAsGroup());
 
+        podTemplate.setHostNetwork(template.isHostNetworkSet() ? template.isHostNetwork() : parent.isHostNetwork());
```

Comment:
It seems this (and/or the `withHostNetwork` above) is the only really critical part: we want to ensure that pods _specified as YAML_ can properly merge all fields supported by Kubernetes. But adding Java accessors and config fields for all of these is not maintainable; we want to encourage users to use YAML, not the `containers` syntax. So I wonder if `combine` could be rewritten to do a YAML/JSON-level structural merge without hard-coding all of these fields, making patches like this unnecessary. @Vlatombe WDYT?

Reply:
I'm sure it could be rewritten (very carefully). It is indeed cumbersome to add support for new fields from the pod spec. Though a large rewrite is certainly outside of the scope of this PR.

Do you want me to remove the code in PodTemplateStep to discourage usage of the pipeline styled fields in podTemplate? (Seems a bit inconsistent though) How do I move this PR along?.

Reply:
Ping @Vlatombe

Reply:
Sure, would be best to have a global approach.

Reply:
I filed [JENKINS-59984](https://issues.jenkins-ci.org/browse/JENKINS-59984) to track the need for a broader change.